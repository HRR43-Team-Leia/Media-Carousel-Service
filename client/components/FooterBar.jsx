import React, { useState } from 'react';
import styled from 'styled-components';


function FooterBar() {
  const [isBlack, setIsBlack] = useState(false);

  const toggle = () => {

  }

  return (
    <div className="footer-bar">
      <p>Footer</p>
    </div>
  )
}

export default FooterBar;