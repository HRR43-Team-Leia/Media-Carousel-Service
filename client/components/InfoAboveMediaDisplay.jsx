import React, { useState, useEffect } from 'react';
import styled from 'styled-components';

const Wrapper = styled.section`
padding: 4em;
background: black;
display: flex;
`;

const experienceHardCodedData = {
  city: 'Mexico City',
  state: null,
  country: 'Mexico',
  category: 'Arts and Culture',
  activity: 'History'
}

function InfoAboveMediaDisplay () {
  const [experienceId, setExperienceId] = useState(experienceHardCodedData);

  // useEffect to perform API call to get data
  // useEffect(() => {

  // });

  return (
    <div>
      <Wrapper>
        <ul className="listCityAndActivity">
          <li className="listItem">
            <a>
              <span className="city-country">Mexico City, Mexico</span>
            </a>
          </li>
          <li className="listItem">
            <span className="slash"> / </span>
          </li>
          <li className="listItem">
            <a>
              <span className="category">Arts and Culture</span>
            </a>
          </li>
          <li className="listItem">
            <span className="slash"> / </span>
          </li>
          <li className="listItem">
            <a>
              <span className="activity">Art Walk</span>
            </a>
          </li>
        </ul>
      </Wrapper>
    </div>
  )

}


export default InfoAboveMediaDisplay;

