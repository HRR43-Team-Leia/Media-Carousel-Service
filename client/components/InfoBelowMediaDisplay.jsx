import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faStar, faClock, faUserFriends, faReceipt, faComments,
} from '@fortawesome/free-solid-svg-icons';

const Wrapper = styled.section`
box-sizing: border-box;
width: 100vw;
background: black;
display: flex;
padding-top: 30px;
padding-bottom: 40px;
`;

const FirstCol = styled.div`
  flex-basis: 35%;
  flex-wrap: wrap;
  justify-content: flex-start;
`;

const SecondCol = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-basis: 65%;
`;

const IndvDetail = styled.div`
display: flex;
flex-direction: column;
flex-basis: 25%;
padding-right: 35px;
`;

const StarsAndReviews = styled.span`
padding-right: 8px;
`;

const DetailsGroup = styled.div`
display: flex;
flex-direction: row;
justify-content: flex-around;
padding: 15px 0 0 0;
`;

class InfoBelowMediaDisplay extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      navbarBottomPosition: null,
    };

    this.listenScrollEvent = this.listenScrollEvent.bind(this);
  }

  componentDidMount() {
    window.addEventListener('scroll', this.listenScrollEvent);
  }

  listenScrollEvent() {
    const navBar = document.querySelector('.infoBelowBottom').getBoundingClientRect();
    this.setState({ navbarBottomPosition: navBar.bottom });
    const { updateNavBarPosition } = this.props;
    const { navbarBottomPosition } = this.state;
    updateNavBarPosition(navbarBottomPosition);
  }

  render() {
    const { languages } = this.props;
    const { includes } = this.props;

    return (
      <Wrapper className="infoBelowBottom">
        <section className="infoBelow">
          <FirstCol>
            <div className="activity-classification">{this.props.experience.activity}</div>
            <div className="activity-title">{this.props.experience.title}</div>
            <div className="activity-city-state">
              {this.props.experience.city}
,
              {' '}
              {this.props.experience.country}
            </div>
            <div className="average-rating">
              <StarsAndReviews>
                {this.props.experience.averageRating}
              </StarsAndReviews>
              <StarsAndReviews>
                <FontAwesomeIcon icon={faStar} size="sm" />
              </StarsAndReviews>
              (
              {this.props.experience.numberOfReviews}
                )
              <StarsAndReviews />
            </div>
            <div>{this.props.experience.tags}</div>
          </FirstCol>
          <SecondCol>
            <DetailsGroup>
              <IndvDetail>
                <div className="duration-icon">
                  <FontAwesomeIcon icon={faClock} size="xs" />
                </div>
                <div className="duration-title-time">
                  <div className="duration-title" style={{ color: '#DCDCDC' }}>Duration</div>
                  <div className="hours">
                    {this.props.experience.duration}
                    {' '}
hours
                  </div>
                </div>
              </IndvDetail>
              <IndvDetail>
                <div className="group-size-icon">
                  <FontAwesomeIcon icon={faUserFriends} size="sm" />
                </div>
                <div className="group-size-title-time">
                  <div className="group-size-title" style={{ color: '#DCDCDC' }}>Group size</div>
                  <div className="group-size">
Up to
                    {' '}
                    {this.props.experience.groupSize}
                    {' '}
people
                  </div>
                </div>
              </IndvDetail>
              <IndvDetail>
                <div className="includes-icon">
                  <FontAwesomeIcon icon={faReceipt} size="sm" />
                </div>
                <div className="includes-title-items">
                  <div className="includes-title" style={{ color: '#DCDCDC' }}>Includes</div>
                  <div className="includes-items">
                    {includes.map((include) => (
                      <div key={include}>
                        {include}
,
                      </div>
                    ))}

                  </div>
                </div>
              </IndvDetail>
              <IndvDetail>
                <div className="hosted-in-icon">
                  <FontAwesomeIcon icon={faComments} size="sm" />
                </div>
                <div className="hosted-in-title">
                  <div className="duration-title" style={{ color: '#DCDCDC' }}>Hosted in</div>
                  <div className="languages">
                    {languages.map((language) => (
                      <div key={language}>
                        {language}
,
                      </div>
                    ))}
                  </div>
                </div>
              </IndvDetail>
            </DetailsGroup>
          </SecondCol>
        </section>
      </Wrapper>
    );
  }
}


export default InfoBelowMediaDisplay;
