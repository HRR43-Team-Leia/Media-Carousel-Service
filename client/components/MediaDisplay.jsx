import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faShareSquare } from "@fortawesome/free-solid-svg-icons";
import { faHeart } from "@fortawesome/free-regular-svg-icons";

const Wrapper = styled.section`
  box-sizing: border-box;
  background: black;
  display: flex;
`;
// width: 100vw;

function MediaDisplay({ images }) {
  return (
    <Wrapper>
      <section className="image-section">
        <div className="image-cover">
          <img src="https://picsum.photos/id/1018/200/200" />
          {/* <img src={images[0]} alt="position 1" /> */}
        </div>
        <div className="image-cover">
          <img src="https://picsum.photos/id/1018/200/200" />
          {/* <img src={images[1]} alt="position 2" /> */}
        </div>
        <div className="vertical-stack-images">
          <div className="image-cover-stacked">
            <img src="https://picsum.photos/id/1018/200/200" />

            {/* <img src={images[2]} alt="position 3" /> */}
          </div>
          <div className="image-cover-stacked">
            <img src="https://picsum.photos/id/1018/200/200" />

            {/* <img src={images[3]} alt="position 4" /> */}
          </div>
        </div>
        <div className="image-cover-last">
          <img src="https://picsum.photos/id/1018/200/200" />

          {/* <img src={images[4]} alt="position 5" /> */}
        </div>
        <div className="share-save">
          <div className="share">
            <button type="button" className="save-btn">
              <FontAwesomeIcon
                className="save-icon"
                icon={faShareSquare}
                size="1x"
              />
              <span className="share-text">Share</span>
            </button>
          </div>
          <div className="save">
            <button type="button" className="share-btn">
              <FontAwesomeIcon
                className="share-icon"
                icon={faHeart}
                size="1x"
              />
              <span className="save-text">Save</span>
            </button>
          </div>
        </div>
      </section>
    </Wrapper>
  );
}

MediaDisplay.propTypes = {
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default MediaDisplay;
