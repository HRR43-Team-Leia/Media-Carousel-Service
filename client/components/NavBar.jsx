import React from 'react';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';

const Wrapper = styled.section`
box-sizing: border-box;
width: 100vw;
background: black;
position: sticky;
top: 0;
display: flex;
justify-content: space-between;
padding: 20px 20px 10px 0;
`;


const LogoAndSearch = styled.div`
display: flex;
flex-direction: row;
justify-content: flex-start;
align-items: center;
`;

const SearchIcon = styled.div`
padding-right: 20px;
`;

const SearchBar = styled.div`
border-radius: 4px;
background: white;
width: 35vw;
padding: 10px 10px 10px 10px;
`;

const SearchInput = styled.input`
padding-left: 8px;
font-size: 14px;
border: white;
`;

const Sections = styled.div`
display: flex;
flex-direction: row;
justify-content: flex-end;
align-items: center;
color: white;
width: 45vw;
flex-wrap: nowrap;
padding: 10px;
`;

const Topics = styled.div`
padding-left: 12px;
display: flex;
flex-wrap: nowrap;
font-size: 12px;
`;

function NavBar() {
  return (
    <Wrapper>
      <LogoAndSearch>
        <SearchIcon>
          <FontAwesomeIcon className="airbnb-icon" icon={['fab', 'airbnb']} size="2x" style={{ color: 'white' }} />
        </SearchIcon>
        <SearchBar>
          <FontAwesomeIcon icon={faSearch} size="1x" style={{ color: 'grey' }} />
          <SearchInput type="text" placeholder="Search" />
        </SearchBar>
      </LogoAndSearch>
      <Sections>
        <Topics>English</Topics>
        <Topics>$ USD</Topics>
        <Topics>Become a Host</Topics>
        <Topics>Sign up</Topics>
        <Topics>Log in</Topics>
      </Sections>
    </Wrapper>
  );
}

export default NavBar;
