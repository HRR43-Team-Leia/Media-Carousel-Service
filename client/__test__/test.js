import React from 'react';
import Enzyme, {
  shallow,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import InfoBelowMediaDisplay from '../components/InfoBelowMediaDisplay';
import MediaDisplay from '../components/MediaDisplay';
import InfoAboveMediaDisplay from '../components/InfoAboveMediaDisplay';
import NavBar from '../components/NavBar';
import FooterBar from '../components/FooterBar';

Enzyme.configure({ adapter: new Adapter() });

describe('Photos And Info Test', () => {
  describe('InfoBelowMediaDisplay should render correctly ', () => {
    const testExperience = {
      includes: ['Et', 'Th'],
      hostedLanguages: ['Tenetur', 'Totam'],
      imageUrls: ['https://fec-media.s3.amazonaws.com/photo17.jpg', 'https://fec-media.s3.amazonaws.com/photo4.jpg', 'https://fec-media.s3.amazonaws.com/photo10.jpg', 'https://fec-media.s3.amazonaws.com/photo15.jpg', 'https://fec-media.s3.amazonaws.com/photo18.jpg'],
      experienceId: 1,
      title: 'Aut Quia Qui',
      city: 'Tessiechester',
      state: 'Maryland',
      country: 'Moldova',
      category: 'Et Placeat Molestiae',
      activity: 'Expedita Quia Sunt',
      averageRating: 2,
      numberOfReviews: 27,
      duration: 6,
      groupSize: 24,
      cuisine: 'Velit',
      costPerPerson: 245,
      videoUrl: 'http://akeem.com',
    };

    it('should be defined', () => {
      expect(InfoBelowMediaDisplay).toBeDefined();
    });

    const Wrapper = shallow(<InfoBelowMediaDisplay experience={testExperience} />);


    it('InfoBelowMediaDisplay should shallow render', () => {
      expect(Wrapper).toMatchSnapshot();
    });
  });

  describe('MediaDisplay should render correctly ', () => {
    const imageTest = ['https://fec-media.s3.amazonaws.com/photo17.jpg', 'https://fec-media.s3.amazonaws.com/photo4.jpg', 'https://fec-media.s3.amazonaws.com/photo10.jpg', 'https://fec-media.s3.amazonaws.com/photo15.jpg', 'https://fec-media.s3.amazonaws.com/photo18.jpg'];

    const Wrapper = shallow(<MediaDisplay images={imageTest} />);

    it('should be defined', () => {
      expect(MediaDisplay).toBeDefined();
    });

    it('MediaDisplay should shallow render', () => {
      // const Wrapper = shallow(<MediaDisplay />);

      expect(Wrapper).toMatchSnapshot();
    });

    it('expect 5 images to be rendered', () => {
      // const wrapper = shallow(<MediaDisplay />);
      expect(Wrapper.find('img')).toHaveLength(5);
    });
  });

  describe('InfoAboveMediaDisplay should render correctly ', () => {
    it('should be defined', () => {
      expect(InfoAboveMediaDisplay).toBeDefined();
    });

    it('InfoAboveMediaDisplay should shallow render', () => {
      const Wrapper = shallow(<InfoAboveMediaDisplay />);

      expect(Wrapper).toMatchSnapshot();
    });
  });

  describe('NavBar should render correctly ', () => {
    it('should be defined', () => {
      expect(NavBar).toBeDefined();
    });

    it('NavBar should shallow render', () => {
      const Wrapper = shallow(<NavBar />);
      expect(Wrapper).toMatchSnapshot();
    });
  });

  describe('FooterBar should render correctly ', () => {
    it('should be defined', () => {
      expect(FooterBar).toBeDefined();
    });

    it('FooterBar should shallow render', () => {
      const Wrapper = shallow(<FooterBar />);
      expect(Wrapper).toMatchSnapshot();
    });
  });
});


// {_id:{"$oid":"5e3b1fdf930faeb939794bf7"},
// includes:["Et"],
// hostedLanguages:["Tenetur","Totam"],
// imageUrls:["https://fec-media.s3.amazonaws.com/photo17.jpg","https://fec-media.s3.amazonaws.com/photo4.jpg","https://fec-media.s3.amazonaws.com/photo10.jpg","https://fec-media.s3.amazonaws.com/photo15.jpg","https://fec-media.s3.amazonaws.com/photo18.jpg"],
// experienceId:{"$numberInt":"1"},
// title:"Aut Quia Qui","city":"Tessiechester",
// state:"Maryland",
// country:"Moldova",
// category:"Et Placeat Molestiae",
// activity:"Expedita Quia Sunt",
// averageRating:{"$numberInt":"2"},
// numberOfReviews:{"$numberInt":"27"},
// duration:{"$numberInt":"6"},
// groupSize:{"$numberInt":"24"},
// cuisine:"Velit",
// costPerPerson:{"$numberInt":"245"},
// videoUrl:"http://akeem.com",
// __v:{"$numberInt":"0"}}

// {
// includes:["Et"],
// hostedLanguages:["Tenetur","Totam"],
// imageUrls:["https://fec-media.s3.amazonaws.com/photo17.jpg","https://fec-media.s3.amazonaws.com/photo4.jpg","https://fec-media.s3.amazonaws.com/photo10.jpg","https://fec-media.s3.amazonaws.com/photo15.jpg","https://fec-media.s3.amazonaws.com/photo18.jpg"],
// experienceId:1,
// title:"Aut Quia Qui","city":"Tessiechester",
// state:"Maryland",
// country:"Moldova",
// category:"Et Placeat Molestiae",
// activity:"Expedita Quia Sunt",
// averageRating:2,
// numberOfReviews:27,
// duration:6,
// groupSize:24,
// cuisine:"Velit",
// costPerPerson: 245,
// videoUrl:"http://akeem.com",
// }


// {_id:{"$oid":"5e3b1fdf930faeb939794bf7"},
// includes:["Et"],
// hostedLanguages:["Tenetur","Totam"],
// imageUrls:["https://fec-media.s3.amazonaws.com/photo17.jpg","https://fec-media.s3.amazonaws.com/photo4.jpg","https://fec-media.s3.amazonaws.com/photo10.jpg","https://fec-media.s3.amazonaws.com/photo15.jpg","https://fec-media.s3.amazonaws.com/photo18.jpg"],
// experienceId:{"$numberInt":"1"},
// title:"Aut Quia Qui","city":"Tessiechester",
// state:"Maryland",
// country:"Moldova",
// category:"Et Placeat Molestiae",
// activity:"Expedita Quia Sunt",
// averageRating:{"$numberInt":"2"},
// numberOfReviews:{"$numberInt":"27"},
// duration:{"$numberInt":"6"},
// groupSize:{"$numberInt":"24"},
// cuisine:"Velit",
// costPerPerson:{"$numberInt":"245"},
// videoUrl:"http://akeem.com",
// __v:{"$numberInt":"0"}}
