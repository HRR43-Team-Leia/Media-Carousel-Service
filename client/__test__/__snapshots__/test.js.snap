// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoAboveMediaDisplay should render correctly  InfoAboveMediaDisplay should shallow render 1`] = `ShallowWrapper {}`;

exports[`MediaDisplay should render correctly  MediaDisplay should shallow render 1`] = `ShallowWrapper {}`;

exports[`NavBar should render correctly  NavBar should shallow render 1`] = `ShallowWrapper {}`;
